### PR TITLE
[RISC-V] coreclr-vm and other directories in coreclr

### PR DESCRIPTION
**Pull request by clamp03**
- Successfully cross-build for RISC-V.
- Run A simple application "helloworld"
- Fail a test in clr.paltest

### CODE REVIEW

**Comment by clamp03** [comment]
@mangod9 Could you review this PR? Actually I don't know why tests failed or cancelled.
If you leave any comment, I will update as possible as I can.
Thank you.

**Comment by mangod9** [comment]
@janvorli as well. Also are you asking about why the superpmi tests are failing or something else?

**Comment by clamp03** [comment]
Actually, I am asking reviews from you and owners to merge the PR. To get review, I tried to pass all tests, but some tests failed and I cannot find the reasons. :)

File: src/coreclr/inc/regdisp.h
Comment (mangod9) [inline src/coreclr/inc/regdisp.h:429]: is this commented out on purpose?

File: src/coreclr/inc/volatile.h
Comment (mangod9) [inline src/coreclr/inc/volatile.h:72]: `RISCV64` was already included here?

File: src/coreclr/vm/codeman.cpp
Comment (mangod9) [inline src/coreclr/vm/codeman.cpp:912]: this looks equivalent to `TARGET_LOONGARCH64` perhaps just keep it a single implementation?

File: src/coreclr/vm/codeman.cpp
Comment (mangod9) [inline src/coreclr/vm/codeman.cpp:1139]: same as above?

File: src/coreclr/vm/codeman.cpp
Comment (mangod9) [inline src/coreclr/vm/codeman.cpp:2300]: is riscv64 supported on windows?

File: src/coreclr/vm/fieldmarshaler.h
Comment (mangod9) [inline src/coreclr/vm/fieldmarshaler.h:95]: is this definition moved here for consistency?

File: src/coreclr/vm/jitinterface.cpp
Comment (mangod9) [inline src/coreclr/vm/jitinterface.cpp:11263]: unnecessary `;`?

File: src/coreclr/inc/regdisp.h
Comment (clamp03) [reply src/coreclr/inc/regdisp.h:429]: Actually, I did on purpose. In RISC-V, S0 is used as FP. (FP is handled in line 442.) I want to keep it as commented.
However, I think it can be removed.

File: src/coreclr/inc/volatile.h
Comment (clamp03) [reply src/coreclr/inc/volatile.h:72]: Oops. Thank you!

File: src/coreclr/vm/codeman.cpp
Comment (clamp03) [reply src/coreclr/vm/codeman.cpp:912]: Registers and assembly instructions of RISC-V and LOONGARCH64 are more similar than ARM. I implemented RISC-V based on LOONGARCH64.

File: src/coreclr/vm/codeman.cpp
Comment (clamp03) [reply src/coreclr/vm/codeman.cpp:2300]: No. And I don't have any plan to support on windows now.

File: src/coreclr/vm/fieldmarshaler.h
Comment (clamp03) [reply src/coreclr/vm/fieldmarshaler.h:95]: This is for resolving a build error.
```
  [ 93%] Building CXX object vm/wks/CMakeFiles/cee_wks_core.dir/__/weakreferencenative.cpp.o
  /usr/bin/riscv64-linux-gnu-ld: ../../dlls/mscordac/libmscordaccore.so: undefined reference to `NativeFieldDescriptor::GetNestedNativeMethodTable() const'
  clang: error: linker command failed with exit code 1 (use -v to see invocation)               
  make[2]: *** [debug/createdump/CMakeFiles/createdump.dir/build.make:308: debug/createdump/createdump] Error 1
```
fieldmarshalre.cpp is not included for createdump.
I resolved this error by putting the implemention into header.

File: src/coreclr/vm/jitinterface.cpp
Comment (clamp03) [reply src/coreclr/vm/jitinterface.cpp:11263]: Sorry. My mistake.

File: src/coreclr/vm/codeman.cpp
Comment (mangod9) [reply src/coreclr/vm/codeman.cpp:912]: hence wondering if they are the same can we remove duplication for now and just use the LoongArch implementation or is it expected to diverge soon?

Comment (clamp03) [reply src/coreclr/vm/codeman.cpp]: Sorry. I misunderstood your review. I updated it.
Thank you!

**Review by gbalykov — COMMENTED** [review]
Quickly looked through half of the files, have some comments

File: src/coreclr/gc/env/gcenv.base.h
Comment (clamp03) [reply src/coreclr/gc/env/gcenv.base.h]: Thank you. Fixed.

File: src/coreclr/inc/clrconfigvalues.h
Comment (clamp03) [reply src/coreclr/inc/clrconfigvalues.h:747]: Thank you. Fixed.

File: src/coreclr/vm/arm64/cgencpu.h
Comment (clamp03) [reply src/coreclr/vm/arm64/cgencpu.h]: Thank you. Fixed.

File: src/coreclr/gcinfo/CMakeLists.txt
Comment (clamp03) [reply src/coreclr/gcinfo/CMakeLists.txt]: Thank you. Fixed.

File: src/coreclr/inc/clrconfigvalues.h
Comment (clamp03) [reply src/coreclr/inc/clrconfigvalues.h:299]: Thank you. Fixed.

File: src/coreclr/inc/cordebuginfo.h
Comment (clamp03) [reply src/coreclr/inc/cordebuginfo.h]: Right! Fixed.

File: src/coreclr/vm/gccover.h
Comment (clamp03) [reply src/coreclr/vm/gccover.h]: Thank you. I chaged to `0x20000000`, `0x20010000` and `0x20020000`.

File: src/coreclr/vm/riscv64/asmhelpers.S
Comment (t-mustafin) [inline src/coreclr/vm/riscv64/asmhelpers.S:412]: As said in `stubs.cpp` and in comment above `gp` is 12, `sp` is 13.

File: src/coreclr/vm/riscv64/asmhelpers.S
Comment (t-mustafin) [inline src/coreclr/vm/riscv64/asmhelpers.S:450]: Same, `gp` <-> `tp`.

**Review by t-mustafin — CHANGES_REQUESTED** [review]

File: src/coreclr/vm/riscv64/asmhelpers.S
Comment (clamp03) [reply src/coreclr/vm/riscv64/asmhelpers.S]: Thank you. Fixed

File: src/coreclr/vm/riscv64/asmhelpers.S
Comment (clamp03) [reply src/coreclr/vm/riscv64/asmhelpers.S]: Thank you. Fixed

**Comment by clamp03** [comment]
Revert `[[VM] Add getRISCV64PassStructInRegisterFlags](https://github.com/dotnet/runtime/pull/82381/commits/cd1ea45de36458a5e3ec66a325bf3e0c4812680d)` and apply to https://github.com/dotnet/runtime/pull/82379 for coherent JIT-EE implementations/consumptions (https://github.com/dotnet/runtime/pull/82379#discussion_r1159548562)

**Comment by clamp03** [comment]
@mangod9 Could you review it? And if it is good, could you please merge it? I fixed all the things requested by the reviewers and waited 50 days. Other two PRs (https://github.com/dotnet/runtime/pull/82379 , https://github.com/dotnet/runtime/pull/82382) are already merged. Actually, I cannot fix all bugs and unimplemented code with just this PR. After the PR is merged, I will continue to fix and my colleagues and community members can help to support RISC-V in coreclr better.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by clamp03** [comment]
> LGTM, thank you!

Thank you for the approval. If it is okay, could you merge this PR? If it needs more approvals or something else, please let me know.

**Comment by clamp03** [comment]
> @clamp03 there are some conflicts, can you please resolve them?

Sure. I resolved and pushed a commit. Thank you!

File: src/coreclr/vm/CMakeLists.txt
Comment (shushanhf) [inline src/coreclr/vm/CMakeLists.txt:885]: @clamp03  why did you destroy the LoongArch64's config?????

Comment (clamp03) [reply src/coreclr/vm/CMakeLists.txt:885]: @shushanhf  I am really sorry. It is just my mistake. I don't want to destroy LoongArch64 at all. Sorry again.

Comment (shushanhf) [reply src/coreclr/vm/CMakeLists.txt:885]: > @shushanhf I am really sorry. It is just my mistake. I don't want to destroy LoongArch64 at all. Sorry again.

shake hands.

**Pull request merged**
